### PR TITLE
feat(application-settings): add general git repository section

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-458)

- Added section Git repository settings
- Added icon for input select
- Added search for input select 

Capture d’écran 2022-08-19 à 08 22 22

---

### Global

- [x] This PR does not introduce any breaking change
- [ ] This PR introduces breaking change(s) and has been labeled as such
- [x] I have found someone to review this PR and pinged him

### Store

- [x] This PR introduces new store changes

### NX

- [x] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [x] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

File: libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx [hunk 1]
```diff
@@ -8,6 +8,12 @@ import { ApplicationEntity } from '@console/shared/interfaces'
 import { AppDispatch, RootState } from '@console/store/data'
 import PageSettingsGeneral from '../../ui/page-settings-general/page-settings-general'
 
+export const buildGitRepoUrl = (provider: string, branch: string): string => {
```
Comment (bdebon) [inline libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx:11]: could be nice to test this kind of very small utility functions

File: libs/shared/interfaces/src/lib/states/repository.interface.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { RepositoryEntity } from '../domain/repository.entity'
+import { DefaultEntityState } from './default-entity-state.interface'
+
+export interface RepositoryState extends DefaultEntityState<RepositoryEntity> {}
```
Comment (bdebon) [inline libs/shared/interfaces/src/lib/states/repository.interface.ts:4]: Maybe here could be nice to add the join relationship with org

File: libs/shared/ui/src/lib/components/buttons/button/button.tsx [hunk 3]
```diff
@@ -26,6 +26,7 @@ export interface ButtonProps {
   style?: ButtonStyle
   iconLeft?: IconEnum | string
   iconRight?: IconEnum | string
+  iconRightClassName?: string
```
Comment (bdebon) [inline libs/shared/ui/src/lib/components/buttons/button/button.tsx:29]: Won't we would like the same for iconLeftClassName for future cases?

File: libs/store/data/src/lib/root.ts [hunk 4]
```diff
@@ -80,7 +90,11 @@ export const initialRootState = (): RootState => ({
     userSignUp: initialUserSignUpState,
   },
   entities: {
-    organization: initialOrganizationState,
+    organization: {
+      organizations: initialOrganizationState,
+      authProvider: initialAuthProviderState,
```
Comment (bdebon) [inline libs/store/data/src/lib/root.ts:95]: One day we will have to decide if we use plural or single for the naming here 😄

**Review by bdebon — COMMENTED** [review]
gg bg

**Comment by bdebon** [comment]
Just spotted this weird behavior on select.
The search box should be sticky? And the auto scroll behavior at the beginning is a little bit annoying. What do you think about it?
https://user-images.githubusercontent.com/6163954/186132293-e89d82c1-0436-44cf-8dd8-bda927a36c99.mp4

File: libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx:11]: Yes, I already tested it you can see it in the file `page-settings-general-feature.spec.tsx` !

**Pull request merged**